skimage/transform/_warps.py

```diff
@@ -778,6 +760,32 @@ def warp(image, inverse_map, map_args={}, output_shape=None, order=1,
 
         if matrix is not None:
             matrix = matrix.astype(np.double)
+
+            if resize:
+                rows, cols = input_shape[:2]
```

Comment:
This resizing operation appears to only apply to 2-D images whereas the `warp` function accepts both 2-D and 3-D images. Passing in a 3-D image with the argument `resize=True` would likely result in an output with the 3rd axis not resized properly.